### PR TITLE
Make CI publish wheel of the project

### PR DESCRIPTION
**Pull request by ncolomer**
This build and publish fletcher project as a wheel.

Successfully installed the wheel as a pip dependency:
```
(base) root@nco-0:/home/jovyan# conda create -y -n test1 python=3.6
...

(base) root@nco-0:/home/jovyan# conda activate test1

(test1) root@nco-0:/home/jovyan# pip install fletcher==0.3.9.dev9+gb0c781d
...
Successfully installed fletcher-0.3.9.dev9+gb0c781d llvmlite-0.32.1 numba-0.46.0 numpy-1.18.4 pandas-0.25.3 pyarrow-0.15.1 python-dateutil-2.8.1 pytz-2020.1 six-1.15.0

(test1) root@nco-0:/home/jovyan# pip freeze
certifi==2020.4.5.1
fletcher==0.3.9.dev9+gb0c781d
llvmlite==0.32.1
numba==0.46.0
numpy==1.18.4
pandas==0.25.3
pyarrow==0.15.1
python-dateutil==2.8.1
pytz==2020.1
six==1.15.0
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,5 +24,5 @@
         "Programming Language :: Python :: 3.7",
     ],
     packages=find_packages(),
-    install_requires=["pandas>=0.24.0", "pyarrow>=0.15.1", "numba", "six"],
+    install_requires=["pandas>=0.24.0", "pyarrow>=0.15.1", "numba==0.46", "six"],
```
Comment (ncolomer) [inline setup.py:27]: This aligns wheel deps with conda meta.yaml

**Comment by ncolomer** [comment]
@marc9595 the project do not build anymore (88 failed tests), do you have an idea why?

cc @krivonogov 

```
FAILED tests/test_base.py::test_reduce_sum - TypeError: unsupported operand t...
FAILED tests/test_numba_integration.py::test_null_count[array0-0] - numba.err...
FAILED tests/test_numba_integration.py::test_isnull[0-array0-expected0] - num...
FAILED tests/test_numba_integration.py::test_isnull[1-array0-expected0] - num...
FAILED tests/test_numba_integration.py::test_str_length[0-array0-expected0]
FAILED tests/test_numba_integration.py::test_str_length[1-array0-expected0]
FAILED tests/test_numba_integration.py::test_str_concat - numba.errors.Typing...
FAILED tests/test_numba_integration.py::test_decode_example - numba.errors.Ty...
FAILED tests/test_numba_integration.py::test_string_builder_simple[data0] - A...
FAILED tests/test_numba_integration.py::test_string_builder_simple[data3] - A...
FAILED tests/test_numba_integration.py::test_string_builder_simple[data4] - A...
FAILED tests/test_pandas_extension.py::TestBaseDtype::test_construct_from_string_another_type_raises[fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseDtype::test_construct_from_string_another_type_raises[fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseDtype::test_construct_from_string_another_type_raises[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseDtype::test_construct_from_string_another_type_raises[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseDtype::test_construct_from_string_another_type_raises[fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_mask_raises[fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_mask_raises[fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_mask_raises[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_mask_raises[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_mask_raises[fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_array_mask[fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_array_mask[fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_array_mask[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_array_mask[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_array_mask[fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_na_treated_as_false[fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_na_treated_as_false[fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_na_treated_as_false[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_na_treated_as_false[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_boolean_na_treated_as_false[fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type0-list]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type0-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type1-list]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type1-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type2-list]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type2-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type3-list]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type3-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type5-list]
FAILED tests/test_pandas_extension.py::TestBaseGetitemTests::test_getitem_integer_with_missing_raises[fletcher_type5-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_groupby_apply_identity[fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_groupby_apply_identity[fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_groupby_apply_identity[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_groupby_apply_identity[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_groupby_apply_identity[fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_in_numeric_groupby[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseGroupbyTests::test_in_numeric_groupby[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask[False-fletcher_type0-boolean-array-na]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask[False-fletcher_type1-boolean-array-na]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask[False-fletcher_type2-boolean-array-na]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask[False-fletcher_type3-boolean-array-na]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask[False-fletcher_type5-boolean-array-na]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_raises[False-fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_raises[False-fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_raises[False-fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_raises[False-fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_raises[False-fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_boolean_array_with_na[False-fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_boolean_array_with_na[False-fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_boolean_array_with_na[False-fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_boolean_array_with_na[False-fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_mask_boolean_array_with_na[False-fletcher_type5]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_array[False-fletcher_type0-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_array[False-fletcher_type1-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_array[False-fletcher_type2-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_array[False-fletcher_type3-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_array[False-fletcher_type5-integer-array]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type0-list-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type0-integer-array-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type0-integer-array-True]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type1-list-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type1-integer-array-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type1-integer-array-True]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type2-list-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type2-integer-array-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type2-integer-array-True]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type3-list-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type3-integer-array-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type3-integer-array-True]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type5-list-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type5-integer-array-False]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_integer_with_missing_raises[fletcher_type5-integer-array-True]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_preserves_views[fletcher_type0]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_preserves_views[fletcher_type1]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_preserves_views[fletcher_type2]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_preserves_views[fletcher_type3]
FAILED tests/test_pandas_extension.py::TestBaseSetitemTests::test_setitem_preserves_views[fletcher_type5]
```

File: setup.py (hunk 1 above)
Comment (marc9595) [reply setup.py:27]: Maybe try pinning pyarrow version for now ? We didn't check if `pyarrow>=0.16` works with our fletcher repo

**Comment by ncolomer** [comment]
@marc9595 @krivonogov we did it 💪 

This is ready for review, can you have a look?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,6 +23,6 @@ requirements:
     - python
     - numba=0.46
     #numba=0.47 has a bug that will be fixed in the next release, see https://github.com/numba/numba/issues/5056.
-    - pandas>=0.24
-    - pyarrow>=0.15.1
+    - pandas>=0.25.3,<1.0
+    - pyarrow>=0.15.1,<0.16
```
Comment (krivonogov) [inline recipe/meta.yaml:27]: 👍

**Comment by ncolomer** [comment]
@krivonogov already did it (see PR's description above) but your comment make me spot that the package was published as `fletcher` and not `tinyclues_fletcher`, thanks! I'll update the name in `setup.py`.

File: setup.py (hunk 1 above)
Comment (ncolomer) [reply setup.py]: Addressed, thanks @marc9595

**Comment by ncolomer** [comment]
`pip install` is ok from a notebook ✅  

```
(base) root@nco-0:/home/jovyan# conda create -y -n test2 python=3.6
...

(base) root@nco-0:/home/jovyan# conda activate test2

(test2) root@nco-0:/home/jovyan# pip install tinyclues_fletcher==0.3.9.dev10+gf986231
...
Successfully installed llvmlite-0.32.1 numba-0.46.0 numpy-1.18.4 pandas-0.25.3 pyarrow-0.15.1 python-dateutil-2.8.1 pytz-2020.1 six-1.15.0 tinyclues-fletcher-0.3.9.dev10+gf986231

(test2) root@nco-0:/home/jovyan# pip freeze
certifi==2020.4.5.1
llvmlite==0.32.1
numba==0.46.0
numpy==1.18.4
pandas==0.25.3
pyarrow==0.15.1
python-dateutil==2.8.1
pytz==2020.1
six==1.15.0
tinyclues-fletcher==0.3.9.dev10+gf986231
```

Thanks guys, merging.

**Comment by ncolomer** [comment]
Generated release `0.3.9`

**Comment by ncolomer** [comment]
[ch17294]

File: recipe/meta.yaml (hunk 2 above)
Comment (tanguycdls) [reply recipe/meta.yaml:27]: @ncolomer @krivonogov any reason to lock <0.16 ? We're rebasing the Tf branch and we're in pyarrow 0.17 there

Comment (krivonogov) [reply recipe/meta.yaml:27]: there were some tests that broke I guess, @marc9595 will try to fix them and bump pyarrow version: https://app.clubhouse.io/oc/story/22783/update-pyarrow-version-to-0-17-1

**Pull request merged**